src/Storages/MergeTree/MergeTreePartition.cpp

```diff
@@ -434,6 +434,44 @@ void MergeTreePartition::create(const StorageMetadataPtr & metadata_snapshot, Bl
     }
 }
 
+void MergeTreePartition::createAndValidateMinMaxPartitionIds(const StorageMetadataPtr & metadata_snapshot, Block block, ContextPtr context)
```

Comment:
Copy of `MergeTreePartition::create`, but also validates if min max partition keys are equal. If they are different, it means the partition can't be created because the data doesn't belong to the same partition.

Reply:
Better to move this comments from github to the code.